src/Pool.ts

```diff
@@ -2,6 +2,8 @@ import { PendingOperation } from './PendingOperation';
 import { Resource } from './Resource';
 import { checkOptionalTime, delay, duration, now, reflect, tryPromise } from './utils';
 import { EventEmitter } from 'events';
+import { AssertionError } from 'assert';
```

Comment:
What is this for?

Reply:
Looks like that is reminder of early phase of implementation, when there was some code to narrow typing. Will remove it.